src/main/java/problems/impl/TransactionalStack.java

```diff
@@ -0,0 +1,115 @@
+package problems.impl;
+
+import problems.exceptions.TransactionRolledbackException;
+
+import java.util.EmptyStackException;
+import java.util.Stack;
+
+public class TransactionalStack<T> {
+    private final Stack<T> stack = new Stack<>();
+    private final Stack<TransactionEvent<T>> transactionEvents = new Stack<>();
+    private Stack<TransactionEvent<T>> rolledbackTransactionEvents = new Stack<>();
```

Comment:
Why do you need to use the public methods? Can't you just call `pop/push` in the underlying stack directly? There's not much code there to reuse.